### PR TITLE
Don't invoke rustup add unless needed.

### PR DESCRIPTION
**Pull request by jsirois**

### CODE REVIEW

**Comment by jsirois** [comment]
This is in response to @kaos's work in #57 to get a Homebrew package set up.

File: package/src/main.rs [hunk 1]
```diff
@@ -403,6 +405,56 @@ impl BuildContext {
...
+                target = self.target
+            );
+            execute(Command::new("rustup").args(["target", "add", &self.target]))?;
+            self.target_prepared = true;
+        }
+        Ok(())
+    }
+
+    fn obtain_ptex(&mut self, dest_dir: &Path) -> Result<PathBuf, Exit> {
+        if self.ptex_repo.is_some() {
```
Comment (jsirois) [inline package/src/main.rs:422]: I was not clever enough to be able use an `if let Some(...) = ...` binding here and satisfy the borrow checker; so this safe, but unsatisfying is_some / unwrap unconditionally a few lines lower combo.

Comment (jsirois) [reply package/src/main.rs]: Got smarter, Cell allows for a partially mutable struct. Fixed.

**Review by kaos — APPROVED** [review]
Thanks 👍🏽

**Pull request merged**